distributed/shuffle/_arrow.py

```diff
@@ -34,8 +34,8 @@ def check_minimal_arrow_version() -> None:
     Raises a ModuleNotFoundError if pyarrow is not installed or an
     ImportError if the installed version is not recent enough.
     """
-    # First version that supports concatenating extension arrays (apache/arrow#14463)
-    minversion = "12.0.0"
+    # First version to implement type promotion for pa.concat_tables (apache/arrow#36846)
+    minversion = "14.0.0"
```

Comment:
This version came out on Nov 1, 2023. Users are unlikely to have it installed in their existing environments, so this PR will create some user pain forcing them to upgrade or manually choose to use task-based shuffling.

Reply:
can we do this on a best effort basis instead of effectively pinning it? There are many users out there who are not affected by this problem. It would be nice if this wasn't a breaking change for them.

Reply:
Yeah I am worried about increasing to 14 by default as well

Reply:
Just pushed a more permissive approach (it's actually more permissive than what we had before). I just didn't want to go through the hassle unless others think that being more backward-compatible is with it here.